app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt

```diff
@@ -318,6 +324,18 @@ interface ProjectViewModel {
                     .compose(bindToLifecycle())
                     .subscribe { this.setActionButtonId.onNext(it) }
 
+            currentProject
+                    .map { getRewardButtonString(it) }
+                    .take(1)
```

Comment:
Is there a reason why you're doing `take(1)` if the project ever emits after the user backs it, the button will never update.